src/run/objective.rs

```diff
@@ -0,0 +1,51 @@
+use serde::Deserialize;
+
+/// Additional fail condition for a level
+#[derive(Deserialize, Clone, Debug)]
+pub enum Objective {
+    /// Objective representing defense of a planet, structure, etc
+    Defense(DefenseData),
+}
+
+/// Tracks data for the defense objective
+#[derive(Deserialize, Clone, Debug)]
+pub struct DefenseData {
+    /// Current defense
+    defense: f32,
```

Comment:
Since this is new code, we should consider making these `usize` if possible since that is fundamentally simpler and eliminates many error cases.